pkg/tls/self_signed_cert_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Package gencert generates self-signed TLS certificates.
+package selfsignedcert
+
+import (
+	"crypto/tls"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenerateSelfSignedCert(t *testing.T) {
+	// Setup temp file paths
+	certPath := "test_cert.pem"
+	keyPath := "test_key.pem"
+
+	// Cleanup any pre-existing files from previous tests
+	defer os.Remove(certPath)
+	defer os.Remove(keyPath)
+
+	// Test case 1: Cert file already exists
+	t.Run("cert file exists", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("Test 1: Cert file exists", func(t *testing.T) {
```

We have a naming convention for test names where they start with `Test <test number>: `.
Makes it easier to identify quickly which test failed if we have test failures